src/net.cpp

```diff
@@ -2340,6 +2340,27 @@ void CConnman::ProcessFixedSeeds(std::chrono::microseconds start)
                 seed_addrs.erase(std::remove_if(seed_addrs.begin(), seed_addrs.end(),
                                                 [&fixed_seed_networks](const CAddress& addr) { return fixed_seed_networks.count(addr.GetNetwork()) == 0; }),
                                  seed_addrs.end());
+                Shuffle(seed_addrs.begin(), seed_addrs.end(), FastRandomContext());
+
+                // Make AddrFetch connections to fixed seeds first. This reduces the
+                // load on the fixed seeds that would otherwise be serving blocks for
+                // many new peers requiring IBD. Try this with multiple fixed seeds for
+                // a better diversity of received addrs and because some may be offline.
+                LogPrintf("Initiating AddrFetch connections to fixed seeds. This might take up to 2 minutes.\n");
+                for (size_t addr_pos = 0; addr_pos < 10; ++addr_pos) {
+                    if (addr_pos >= seed_addrs.size()) {
+                        break;
+                    }
+                    AddAddrFetch(seed_addrs.at(addr_pos).ToStringAddr());
+                }
+                // Give AddrFetch peers some time to provide us with addresses
+                // before adding the fixed seeds to AddrMan
+                if (!interruptNet.sleep_for(std::chrono::minutes(2))) {
+                    return;
+                }
+                // The fixed seeds queried in the previous steps might have been offline,
+                // failed to send us any addresses or sent us fake ones. Therefore,
+                // we now add all reachable fixed seeds to AddrMan as a fallback.
```

Comment:
dfd635bcc9b3f3615cabe115af302df33efba6a1

nit: The first sentence talks about `might`, while the second makes a reader think this is unconditionally a fallback (which is not true). I'd suggest aligning the two

Reply:
I don't quite follow: The fixed seeds will eventually be added to addrman as a fallback unconditionally, both in the case where we got sufficient addresses from addrfetch peers and in the case where that didn't work for some reason.

Reply:
Ahhh I guess I thought it means "as a fallback to the situation from the previous sentence (poor in-place seeds)", but you mean "as a fallback for a general failure" (not that related to the thing above).

This is an even more subtle nit now :)

Reply:
I have reordered the sentence now so that it is clearer now hopefully!